packages/components/src/lib/queries/lexical-query.js

```diff
@@ -166,30 +167,45 @@ export default class LexicalQuery extends Query {
         if (this._annotatedHomonyms && this._annotatedHomonyms.hasHomonyms) {
           this.homonym = Homonym.disambiguate(this.homonym, this._annotatedHomonyms.homonyms)
         }
-        LexicalQuery.evt.MORPH_DATA_READY.pub()
+        // Suppress events that will trigger UI messages if source is wordlist
+        if (this._source !== LexicalQuery.sources.WORDLIST) {
+          LexicalQuery.evt.MORPH_DATA_READY.pub()
+        }
```

Comment:
The following events - HOMONYM_READY, SHORT_DEFS_READY, LEMMA_TRANSL_READY
executes update provider service, and we save provider information to IndexDB
But I am not sure , that we retrieve that information correctly

Reply:
This is a good catch! If we need to update providers during wordlist requests then we should probably take provider retrieval code out of a UI controller (which would be a good thing from an architectural standpoint, on my opinion). 

Right now callbacks of those events in a UI controller update both the UI and retrieve provider data. We could change it so that a UI controller will be responsible for updating UI only and register another subscriber to those events in probably LexisCS that will be responsible for getting provider data.

The question is how critical is that. I'm afraid to do such changes before a release, it would affect too many parts. If we need it, maybe we can create an issue and do it after the release?  @balmas, what do you think?

Reply:
I agree with your suggestions on refactoring and agree it can wait. The provider information is not really necessary for the wordlist download and in any event, I think what the UI Controller code does is dedupe the provider information that is already there, so at the worst we end up with repetitive information.  Please do enter an issue for addressing this as you suggest in a future release though. Thank you both!